docs/_posts/2024-07-18-Run-your-pandas-workloads-on-cluster.md

```diff
@@ -0,0 +1,202 @@
+(using multiple nodes) or vertically (using a more powerful machine). pandas, limited to a single node,
+can't scale horizontally. Fortunately, Modin on Ray can distribute the workload across multiple nodes,
+making horizontal scaling possible.
+
+### Prerequisites
+
+Modin makes it a breeze to run your existing pandas code in a cluster with minimal changes.
+You can simply submit your Python script as a Ray job to a Ray cluster. Just keep in mind a couple of prerequisites:
+
+1.  You need to have a Ray cluster up and running.
+2.  The dataset must be available (not just accessible) on all nodes in the cluster.
+
+#### Setting Up Ray Cluster and Distributing the Dataset
+
+There are several ways to set up a Ray cluster. You can opt for an on-premise setup or
+take the easier route of configuring a cluster in AWS using tools like
+[KubeRay](https://github.com/ray-project/kuberay/tree/master/helm-chart/ray-cluster).
+In our configuration, we used [init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/)
+to download and prepare data on all nodes of the Ray cluster, ensuring that both prerequisites are effectively met.
+
+Once the Ray cluster is up and running, set the  `RAY_ADDRESS` environment variable
+to the Ray dashboard URL on your client machine.
+
+```bash
+export RAY_ADDRESS=URL_OF_RAY_DASHBOARD
+```
+
+Now, make minor changes to your Python script by importing pandas from Modin and initializing Ray.
+
+```python
+import modin.pandas as pd
+import ray
+ray.init(address="auto", logging_level="WARNING")
+# your workload without any changes
+```
+
+
+
+You can view the complete Python script below.
+
+<details>
+  <summary>modin_workload.py</summary>
+
+    <pre><code class="language-python">
+    # import pandas as pd
+    import modin.pandas as pd
+    import ray
+    from modin.utils import execute # execute is just used to make sure all the asynchronous operations are complete as we benchmark.
+    ray.init(address="auto", logging_level="WARNING")
+    file_path = "/home/ray/data/big_yellow.csv"
+    df = pd.read_csv(file_path)
+    df = df.map(str)
+    result = pd.DataFrame()
+
+    payment_type_map = {
+        1: 'Credit Card',
+        2: 'Cash',
+        3: 'No Charge',
+        4: 'Dispute',
+        5: 'Unknown',
+        6: 'Voided Trip'
+    }
+
+    result["df_payment_type"] = df['payment_type'].map(payment_type_map)
+
+    def custom_function(row):
+        return str(row["passenger_count"]) + " passengers were picked up at " + str(row["tpep_pickup_datetime"])
+
+    result["description"] = df.apply(custom_function, axis=1)
+    execute(result)
+    </code></pre>
+
+  ```
+</details>
+
+### Results
+
+
+We can observe that by increasing the number of nodes in the cluster, Modin could handle the workload
+effectively with 4 nodes or more. Significant performance improvements were evident as we scaled
+from 4 to 32 nodes. Given that Modin operates efficiently with 4 nodes and both pandas and Modin
```

Comment:
Is this correct to talk about pandas as to number of nodes?

Reply:
would it be okay if we mention in brackets (as pandas does not offer horizontal scaling)?

Reply:
Yes, let's put this clarification.

Reply:
Added clarification